example/lib/pages/custom_formatters.dart

```diff
@@ -0,0 +1,221 @@
+                    'day': Variable(
+                      accessor: (_MyCustomData map) => map.day,
+                    ),
+                    'price': Variable(
+                      accessor: (_MyCustomData map) => map.price,
+                    ),
+                    'stock': Variable(
+                      accessor: (_MyCustomData map) => map.stock,
+                    ),
+                  },
+                  elements: [LineElement(), LineElement()],
+                  axes: [
+                    Defaults.horizontalAxis,
+                    Defaults.verticalAxis,
+                  ],
+                  selections: {
+                    'hover': PointSelection(
+                      dim: Dim.x,
+                      on: {GestureType.hover},
+                      clear: {GestureType.mouseExit},
+                    )
+                  },
+                  tooltip: TooltipGuide(
+                    tooltipFieldFormatter: ({
+                      required field,
+                      required scale,
+                      required selectedIndex,
+                      required selectedTuple,
+                    }) {
+                      if (field == 'price') {
+                        return _kData[selectedIndex].formattedPrice;
+                      }
+
+                      return scale.format(selectedTuple[field]) ?? 'what?';
+                    },
+                  ),
+                  crosshair: CrosshairGuide(),
+                ),
+              ),
+              Container(
+                margin: const EdgeInsets.only(top: 10),
+                width: 350,
+                height: 300,
+                child: Chart(
+                  data: _kData,
+                  variables: {
+                    'day': Variable(
+                      accessor: (_MyCustomData map) => map.day,
+                    ),
+                    'price': Variable(
+                      accessor: (_MyCustomData map) => map.price,
+                    ),
+                    'stock': Variable(
+                      accessor: (_MyCustomData map) => map.stock,
+                    ),
+                  },
+                  elements: [LineElement(), LineElement()],
+                  axes: [
+                    AxisGuide(
+                      line: StrokeStyle(color: const Color(0xFF000000)),
+                      label: LabelStyle(
+                          offset: const Offset(0, 7.5),
+                          span: (value) {
+                            final parsedValue =
+                                double.tryParse(value)?.toStringAsFixed(0) ??
+                                    value;
+                            return TextSpan(
+                              text: 'day $parsedValue',
+                              style: const TextStyle(
+                                color: Color(0xFF000000),
+                                fontSize: 10,
+                              ),
+                            );
```

Comment:
I'll probably need the index here as well, so I can check my tuples for the formatted value